js/latoken.js

```diff
@@ -0,0 +1,715 @@
+        return this.parseOrder(response)
+    }
+    
+    parseAllOrders(orders) {
+        let pairId = orders['pairId'];
+        let symbol = orders['symbol']
+        let canceledOrders = [];
+        for (let i = 0; i < orders.cancelledOrders.length; i++) {
+            canceledOrders.push(
+                orders.cancelledOrders[i]
+            )
+        }
+        let result = {
+            'pairId': pairId,
+            'symbol': symbol,
+            'cancelledOrders': canceledOrders,
+        }
+        return result;
+    }
+
+    async cancelAllOrders(symbol, params = {}) {
+        await this.loadMarkets();
+        let response =  await this.privatePostOrderCancelAll(this.extend({
+            'symbol': symbol
+        }, params));
+        return this.parseAllOrders(response);
+    }
+
+    sign (path, api = 'public', method = 'GET', params = undefined, headers = undefined, body = undefined) {
+        let signature;
+        let query1;
+        let url = this.urls['api']['v1'];
+        url += '/' + path;
+        if (api === 'public') {
+            headers = {
+                'Content-type': 'application/json',
+                'x-lat-timestamp': this.nonce(),
+                'x-lat-timeframe': this.options['timeframe'],
+            };
+            if ((path === 'exchangeInfo/pairs' || 'exchangeInfo/currencies') && (typeof(params) === 'string')) {
+                url += '/' + params;                
+            }
+            url += '?' + this.urlencode (params);
+        } else  if (api === 'private') {
+            this.checkRequiredCredentials();
+            if (path === 'account/balances' && (typeof(params) === 'string')) {
+                url += '/' + params;
+                let param = {
+                    'timestamp': Date.now(),
+                }
+                query1 = `?${Object.entries(param).map(([key, val]) => `${key}=${val}`).join('&')}`
+                let dataToSign = '/api/v1/' + path + '/' + params; 
+                signature = crypto.createHmac('sha256', this.secret).update(dataToSign+query1).digest('hex');
+            } else {
+                params['timestamp'] = this.nonce();
+                query1 = `?${Object.entries(params).map(([key, val]) => `${key}=${val}`).join('&')}`
+                let dataToSign = '/api/v1/' + path; 
+                signature = crypto.createHmac('sha256', this.secret).update(dataToSign+query1).digest('hex');
+            }
+            headers = {
+                'X-LA-KEY': this.apiKey,
+                'X-LA-SIGNATURE': signature,
+            };
+            body = this.urlencode(params);
+            url += query1;
+        }
+        return {'url': url, 'method': method, 'body': body, 'headers': headers};
+    }
+
+    handleErrors (body, response) {
+        const msg = this.safeString (response, 'error');
+        const success = this.safeValue (response, 'success', true);
+        if (msg !== undefined) {
+            throw new ExchangeError (msg);
```

Comment:
We need to add proper exception handling and differentiate exceptions (`DDoSProtection` from `InsufficientFunds`, etc), as it is done, for example, here: 

- https://github.com/ccxt/ccxt/blob/master/js/bitfinex.js#L309
- https://github.com/ccxt/ccxt/blob/master/js/bitfinex.js#L1106

Reply:
done